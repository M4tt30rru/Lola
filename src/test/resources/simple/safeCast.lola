##Import "std.lola"
##Find ##Type(_type) ? (##Expression(_exp))
 ##replace Optional.ofNullable(##(_exp)).filter(__ -> __ instanceof ##(_type)).orElse(null)

Dog ? (f().g().h)
