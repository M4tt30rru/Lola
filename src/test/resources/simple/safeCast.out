Optional.ofNullable(f().g().h).filter(__ -> __ instanceof Dog).orElse(null)
